File: src/main/java/handlers/FeedbackHandler.java
package handlers;

import javafx.animation.FadeTransition;
import javafx.animation.PauseTransition;
import javafx.scene.paint.Color;
import javafx.scene.text.Text;
import javafx.util.Duration;

public class FeedbackHandler {
    private final Text feedbackText;

    public FeedbackHandler(Text feedbackText) {
        this.feedbackText = feedbackText;
    }

    public void displaySuccess(String spellName) {
        feedbackText.setText("Spell saved as " + spellName);
        feedbackText.setFill(Color.GREEN);
        fadeOutFeedbackText();
    }

    public void displayError() {
        feedbackText.setText("Failed to save Spell. Check log for further information.");
        feedbackText.setFill(Color.RED);
        fadeOutFeedbackText();
    }

    public void pointOutIncompetence() {
        feedbackText.setText("Failed to save Spell. Check your inputs and try again.");
        feedbackText.setFill(Color.RED);
        fadeOutFeedbackText();
    }

    private void fadeOutFeedbackText() {
        FadeTransition fadeTransition = new FadeTransition(Duration.seconds(1), feedbackText);
        fadeTransition.setFromValue(1.0);
        fadeTransition.setToValue(0.0);

        PauseTransition pauseTransition = new PauseTransition(Duration.seconds(1.5));
        pauseTransition.setOnFinished(event -> fadeTransition.play());

        pauseTransition.play();
    }

    public Text getFeedbackText() {return feedbackText;}
}

File: src/main/kotlin/idiotamspielen/vttproject/handlers/FeedbackHandler.kt
package idiotamspielen.vttproject.handlers

import javafx.animation.FadeTransition
import javafx.animation.PauseTransition
import javafx.event.ActionEvent
import javafx.event.EventHandler
import javafx.scene.paint.Color
import javafx.scene.text.Text
import javafx.util.Duration

class FeedbackHandler(private val feedbackText: Text) {

    enum class FeedbackType(val color: Color) {
        SUCCESS(Color.GREEN),
        ERROR(Color.RED),
        WARNING(Color.ORANGE),
        INFO(Color.BLUE)
    }

    fun displayFeedback(message: String, type: FeedbackType) {
        feedbackText.text = message
        feedbackText.fill = type.color

        fadeOutFeedbackText()
    }

    private fun fadeOutFeedbackText() {
        val fadeTransition = FadeTransition(Duration.seconds(1.0), feedbackText)
        fadeTransition.fromValue = 1.0
        fadeTransition.toValue = 0.0

        val pauseTransition = PauseTransition(Duration.seconds(1.0))
        pauseTransition.onFinished = EventHandler { event: ActionEvent? -> fadeTransition.play() }

        pauseTransition.play()
    }
}
